Microsoft.DotNet.Interactive/IExtensibleKernel.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using MLS.Agent.Tools;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Interactive
+{
+    public interface IExtensibleKernel
+    {
+        public Task LoadExtensionsInDirectory(IDirectoryAccessor directory, KernelInvocationContext invocationContext);
```

Comment:
Will there always be an invocation context? What about extensions that we want to load on startup?

Reply:
We need to "Publish" an event saying that the extension has been loaded and we need the context for that. If we would like to load extensions at startup, how would we "Publish" that event if we dont have the KernelInvocationContext ?

I think once we have that scenario, it would not be a hard change to make this method maybe return a list of extensions that it loaded(so that it can be called at startup) and then the caller can do whatever necessary. Thoughts ?